road_edge_departure_vehicle.py

```diff
@@ -25,29 +26,30 @@
 weather = [[0.0, 0.0, 0.0], [1.0, 0.0, 1.0], [0.0, 1.0, 0.0]]   #[rain, fog, wetness]
 
 #variables
-parkingSpot = [150, 10]  #the forward, right position for parking spot
-npcSpot = [130, 0]  #the forward, right position for npcSpot spot
-npcSpeed = 4    #npcSpeed in terms of forward
-egoSpeed = 2
+ego_position = [150, 10]  #the forward, right position for parking spot
+npc_position = [130, 0]  #the forward, right position for npcSpot spot
+npc_speed = 4    #npcSpeed in terms of forward
+ego_speed = 2
 
 
 for time in time_of_day:
   for weatherInfo in weather:
 
-    #npc
-    state = copy.deepcopy(lgsvl.AgentState())
-    state.velocity = npcSpeed * forward 
-    state.transform = copy.deepcopy(spawns[0])
-    state.transform.position = spawns[0].position + npcSpot[0]*forward + npcSpot[1]*right
-    npc = sim.add_agent("Sedan", lgsvl.AgentType.NPC, state)
-
-    #ego
-    state = copy.deepcopy(lgsvl.AgentState())
-    state.velocity = egoSpeed * right *  -1
-    state.transform = copy.deepcopy(spawns[0])
-    state.transform.position = spawns[0].position + (parkingSpot[0])*forward + (parkingSpot[1])*right
-    state.transform.rotation.y = spawns[0].rotation.y + 270.0
-    ego = sim.add_agent("Lincoln2017MKZ (Apollo 5.0)", lgsvl.AgentType.EGO, state)
+    #ego vehicle
+    ego_vehicle = vehicle.Vehicle(lgsvl.AgentState(), sim)
+    ego_vehicle.set_direction(-right)
+    ego_vehicle.set_velocity(ego_speed)
+    ego_vehicle.set_position(ego_position)
+    ego_vehicle.set_rotation_y(270.0)
+    ego = ego_vehicle.load_vehicle("Lincoln2017MKZ (Apollo 5.0)", lgsvl.AgentType.EGO)
```

Comment:
I do not know what are you trying to do here. The new API makes me confusing. We want an API that is easy to understand without domain knowledge right? So we can have a function to SetState() with some DEFAULT valuable, so the user does not have to set the state every single time. Instead, YOU set the DEFAULT available for them.

So for Line 39 to 44 can be simplified as state = vehicle.setOrGetDefaultState(...)
npc = sim.add_agent("Sedan", lgsvl.AgentType.NPC, state)

No need for load_vehicle